python/pyspark/sql/utils.py

```diff
@@ -61,6 +61,8 @@ def deco(*a, **kw):
                                              e.java_exception.getStackTrace()))
             if s.startswith('org.apache.spark.sql.AnalysisException: '):
                 raise AnalysisException(s.split(': ', 1)[1], stackTrace)
+            if s.startswith('org.apache.spark.sql.catalyst.analysis.NoSuchTableException: '):
```

Comment:
The failed test case is https://github.com/apache/spark/blob/29314379729de4082bd2297c9e5289e3e4a0115e/python/pyspark/sql/tests.py#L1564-L1567

I checked the history. The reason why we captured the exceptions is for issuing a better error message. See the original PR: https://github.com/apache/spark/pull/7135

I will try to add all the exception types. Thanks!